microsoft-edge/webview2/howto/chromium-devtools-protocol.md

```diff
@@ -0,0 +1,190 @@
+    
+    ```csharp
+    webView.CoreWebView2.Navigate(@"C:\{PATH TO FILE}\geolocater.html");
+    ```  
+    
+1.  Ensure the `geolocator.html` file displays in your WebView2 control app.  
+    
+    :::image type="complex" source="./media/initial-geolocate.png" alt-text="Display the geolocater.html file in your WebView2 control app" lightbox="./media/initial-geolocate.png":::
+       Display the `geolocater.html` file in your WebView2 control app  
+    :::image-end:::  
+    
+## Step 3: Install the DevToolsProtocolHelper NuGet package  
+
+Use NuGet to download `Microsoft.Web.WebView2.DevToolsProtocolExtension`.  To install the package, complete the following actions.  
+
+1.  Open Visual Studio.  
+1.  Choose **Project** > **Manage NuGet Packages** > **Browse** > **Include pre-release**.
+1.  Type `Microsoft.Web.WebView2.DevToolsProtocolExtension` and choose **Microsoft.Web.WebView2.DevToolsProtocolExtension** > **Install**.  
+    
+> [!NOTE] 
+> Ensure you also have the WebView2 NuGet Package named `Microsoft.Web.WebView2` installed.  
+
+:::image type="complex" source="./media/cdpnuget.png" alt-text="Ensure Microsoft.Web.WebView2.DevToolsProtocolExtension displays in the Visual Studio NuGet Package Manager" lightbox="./media/cdpnuget.png":::
+   Ensure **Microsoft.Web.WebView2.DevToolsProtocolExtension** displays in the Visual Studio NuGet Package Manager  
+:::image-end:::  
+
+## Step 4: Use DevTools Protocol Helper  
+
+1.  Add the `DevToolsProtocolExtension` and `CoreWebView2` namespaces to your project.
+    
+    ```csharp
+    using Microsoft.Web.WebView2.Core;
+    using Microsoft.Web.WebView2.Core.DevToolsProtocolExtension;
+    ```  
+    
+1.  Run the `GetDevToolsProtocolHelper()` method to instantiate the `DevToolsProtocolHelper` object, and navigate to `geolocater.html`.
+    
+    ```csharp
+    async void InitializeAsync()
+    {
+        await webView.EnsureCoreWebView2Async(null);
+        DevToolsProtocolHelper helper = webView.CoreWebView2.GetDevToolsProtocolHelper(); 
+        
+        webView.CoreWebView2.Navigate(@"C:\{PATH TO FILE}\geolocater.html");
+    }
+    ```  
+    
+1.  Run the [setGeoLocationOverrideAsync][setGeoLocationOverrideAsync] method.  For more information, navigate to [method-setGeolocationOverride][GithubChromedevtoolsDevtoolsProtocolTotEmulationMethodSetgeolocationOverride].  
+    
+    ```csharp
+    async void InitializeAsync()
+    {
+        await webView.EnsureCoreWebView2Async(null);
+        DevToolsProtocolHelper helper = webview.CoreWebView2.GetDevToolsProtocolHelper();
+        
+        webView.CoreWebView2.Navigate(@"C:\{PATH TO FILE}\geolocater.html");
+        
+        // Latitude and longitude for Paris, France.
+        double latitude = 48.857024082572565;  
+        double longitude = 2.3161581601457386;  
+        double accuracy = 1;
+        await helper.Emulation.setGeolocationOverrideAsync(latitude, longitude, accuracy);
```

Comment:
Is there any way to work in a CDP event into the code sample too?

Reply:
I couldn't find one that fit nicely/was worried about making it too long. We do have exampels in a CDP sample app Monica is publishing.